#########################################################################
Date: Feb 26, 2024 13:19 PM
Source: Comcast
Upstream-Status: Pending
Signed-off-by: Srikanth Simgamsetti <Srikanth_Simgamsetti@comcast.com>
#########################################################################

Index: bluez-5.80/lib/uuid.c
===================================================================
--- bluez-5.80.orig/lib/uuid.c
+++ bluez-5.80/lib/uuid.c
@@ -113,7 +113,9 @@ int bt_uuid128_create(bt_uuid_t *btuuid,
 int bt_uuid_cmp(const bt_uuid_t *uuid1, const bt_uuid_t *uuid2)
 {
 	bt_uuid_t u1, u2;
-
+if (!uuid2 || uuid2->type == BT_UUID_UNSPEC) {
+	           return -EINVAL;
+		    }
 	bt_uuid_to_uuid128(uuid1, &u1);
 	bt_uuid_to_uuid128(uuid2, &u2);
 
