#########################################################################
Date: Dec 20, 2024 13:19 PM
Source: Sky
Upstream-Status: Pending
Signed-off-by: Ananth Marimuthu <ananth_marimuthu2@comcast.com>
#########################################################################

Index: bluez-5.80/src/shared/shell.c
===================================================================
--- bluez-5.80.orig/src/shared/shell.c
+++ bluez-5.80/src/shared/shell.c
@@ -719,7 +719,7 @@ void bt_shell_printf(const char *fmt, ..
 		saved_line = rl_copy_text(0, rl_end);
 		if (!data.saved_prompt)
 			rl_save_prompt();
-		rl_clear_visible_line();
+		//rl_clear_visible_line();
 		rl_reset_line_state();
 	}
 
