diff --git a/src/daemon/pipewire.conf.in b/src/daemon/pipewire.conf.in
index 27051d66e..071a787ea 100644
--- a/src/daemon/pipewire.conf.in
+++ b/src/daemon/pipewire.conf.in
@@ -12,7 +12,7 @@ context.properties = {
     ## Configure properties in the system.
     #library.name.system                   = support/libspa-support
     #context.data-loop.library.name.system = support/libspa-support
-    #support.dbus                          = true
+    support.dbus                          = true
     #link.max-buffers                      = 64
     link.max-buffers                       = 16                       # version < 3 clients can't handle more
     #mem.warn-mlock                        = false

