diff --git a/src/daemon/systemd/system/pipewire.service.in b/src/daemon/systemd/system/pipewire.service.in
index dc8db3f8f..8329dac6f 100644
--- a/src/daemon/systemd/system/pipewire.service.in
+++ b/src/daemon/systemd/system/pipewire.service.in
@@ -23,12 +23,14 @@ SystemCallArchitectures=native
 SystemCallFilter=@system-service
 Type=simple
 AmbientCapabilities=CAP_SYS_NICE
 ExecStartPre=/bin/mkdir -p /tmp/pipewire
 ExecStart=@PW_BINARY@
 Restart=on-failure
 RuntimeDirectory=pipewire
 RuntimeDirectoryPreserve=yes
 Environment=PIPEWIRE_RUNTIME_DIR=/tmp/pipewire
 Environment=PIPEWIRE_DEBUG=D
 Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/var/run/dbus/system_bus_socket
 
 [Install]
 Also=pipewire.socket pipewire-manager.socket
