diff --git a/src/config/bluetooth.lua.d/50-bluez-config.lua b/src/config/bluetooth.lua.d/50-bluez-config.lua
index c3c2e288..1cdd394b 100644
--- a/src/config/bluetooth.lua.d/50-bluez-config.lua
+++ b/src/config/bluetooth.lua.d/50-bluez-config.lua
@@ -27,7 +27,7 @@ bluez_monitor.properties = {
 
   -- HFP/HSP backend (default: native).
   -- Available values: any, none, hsphfpd, ofono, native
-  --["bluez5.hfphsp-backend"] = "native",
+  ["bluez5.hfphsp-backend"] = "none",
 
   -- Properties for the A2DP codec configuration
   --["bluez5.default.rate"] = 48000,
@@ -76,10 +76,10 @@ bluez_monitor.rules = {
       -- profiles have connected. Disabled by default if the property
       -- is not specified.
       --["bluez5.auto-connect"] = "[ hfp_hf hsp_hs a2dp_sink hfp_ag hsp_ag a2dp_source ]",
-      ["bluez5.auto-connect"]  = "[ hfp_hf hsp_hs a2dp_sink ]",
+      ["bluez5.auto-connect"]  = "[ a2dp_sink ]",
 
       -- Hardware volume control (default: [ hfp_ag hsp_ag a2dp_source ])
-      --["bluez5.hw-volume"] = "[ hfp_hf hsp_hs a2dp_sink hfp_ag hsp_ag a2dp_source ]",
+      ["bluez5.hw-volume"] = "[ a2dp_sink ]",
 
       -- LDAC encoding quality
       -- Available values: auto (Adaptive Bitrate, default)
