diff --git a/src/systemd/system/wireplumber.service.in b/src/systemd/system/wireplumber.service.in
index e7e3751b..448e8b1e 100644
--- a/src/systemd/system/wireplumber.service.in
+++ b/src/systemd/system/wireplumber.service.in
@@ -14,9 +14,13 @@ Type=simple
 AmbientCapabilities=CAP_SYS_NICE
 ExecStart=@WP_BINARY@
 Restart=on-failure
-User=pipewire
-Environment=PIPEWIRE_RUNTIME_DIR=%t/pipewire
+Environment=PIPEWIRE_RUNTIME_DIR=/tmp/pipewire
+Environment=PIPEWIRE_DEBUG=D
+Environment=WIREPLUMBER_DEBUG=4
 Environment=GIO_USE_VFS=local
+Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/var/run/dbus/system_bus_socket
+StandardOutput=file:/tmp/wireplumber.log
+StandardError=file:/tmp/wireplumber.log
 
 [Install]
 WantedBy=pipewire.service
